### PR TITLE
feat(frontend): support alter rate limit in mv, source, table with source

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support frontend parts for altering rate limit with source, source with table, mv.
Only source, source with table now fully support for `ALTER SET STREAMING_RATE_LIMIT TO N`. Look at the slt test for more information.

## Changes for MV (will be done in a future PR)

For MV, this is not fully supported yet. The reason is that we need specific information to identify an MV besides its name (which alter mv provides). We need the schema, and database as well.

These parameters can be provided from the catalog in meta. Currently, when the MV is in `Creating` state, it is persisted on the meta side. However, we currently don't propagate it back to frontend, to make it "invisible" until backfill completes.
We can propagate it back to frontend, and update catalog accessors to mask `Creating` catalogs.

Then we can then access these `Creating` catalogs in frontend to get the necessary info to alter rate limit.

In this PR, the behaviour of MV being backfilled is to reject the alter rate limit command, and inform the user that its catalog is not found.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Supports `ALTER STREAMING_RATE_LIMIT SET = default | N` for source, table with source.

Example:
```
create table kafka_source (v1 int) with (
  connector = 'kafka',
  topic = 'kafka_source',
  properties.bootstrap.server = 'localhost:29092',
  scan.startup.mode = 'earliest',
  streaming_rate_limit = 0
) FORMAT PLAIN ENCODE JSON

alter table kafka_source set streaming_rate_limit to default;

alter table kafka_source set streaming_rate_limit to 1000;
```

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Bump, please leave a review.

**Comment by kwannoel** [comment]
Just found 2 bugs.
1. Table rate limit does not get altered.
2. Source rate limit is not applied initially.

Investigating.

**Comment by kwannoel** [comment]
Depends on https://github.com/risingwavelabs/risingwave/pull/16472

**Comment by kwannoel** [comment]
Ready for review again.

**Review by tabVersion — APPROVED** [review]

File: e2e_test/source/basic/alter/rate_limit_source_kafka.slt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    topic = 'kafka_source',
+    type = 'append-only',
+    force_append_only='true',
+    connector = 'kafka'
+);
+
+############## Source from kafka (rate_limit = 0)
+
+# Wait for the topic to create
+skipif in-memory
```
Comment (xxchan) [inline e2e_test/source/basic/alter/rate_limit_source_kafka.slt:25]: Why to skipif in-memory? :eyes:

**Pull request merged**